src/cargo/ops/cargo_compile.rs

```diff
@@ -74,6 +76,23 @@ pub enum CompileMode {
     Doc { deps: bool },
 }
 
+#[derive(Clone, Copy, PartialEq, Eq)]
+pub enum MessageFormat {
+    Human,
+    Json
+}
+
+impl MessageFormat {
```

Comment:
You can probably get away from directly implementing `rustc_serialize::Decodable` here as that'll thread all the way through docopt I think.